### ISSUE TITLE
Improve dialog and pop-over accessibility

### ISSUE THREAD

**Issue by dcalhoun** [wordpress-mobile/GutenbergKit#86]
## What problem does this address?

When opening a dialog (e.g., block inserter, block settings) or pop-over menu (e.g., block alignment, block actions), screen readers can move focus outside of the element group. This decreases the usability of the interface, as it is difficult to comprehend the current state of the UI. We should ensure that UI outside of the currently displayed dialog or pop-over is hidden from screen readers. 

It appears the lack of focus trapping in relation to HTML UI may be an issue within core Gutenberg. 

## What is your proposed solution?

Done is: 

* When a dialog/pop-over is open, focus cannot be moved to other HTML UI (e.g., block canvas content). 
* When a dialog/pop-over is open, focus cannot be moved to other native UI (e.g., top navigation).

**Comment by dcalhoun** [comment]
> When a dialog/pop-over is open, focus cannot be moved to other native UI (e.g., top navigation).

In #145, I opted to postpone marking native UI as inert, as the current WebView modal dialogs do not visually obscure the native UI. Long term, we might consider updating the how WebView modal dialogs appears visually, and update the screen reader experience then.
